[1m=================================================================================================== test session starts ====================================================================================================
platform linux -- Python 3.9.9, pytest-6.2.2, py-1.11.0, pluggy-0.13.1 -- /home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/bin/python
cachedir: .pytest_cache
rootdir: /home/marc/development/python/udacity_course/genre_classification/starter/check_data
[1mcollected 4 items                                                                                                                                                                                                          
test_data.py::test_column_presence_and_type [31mERROR
test_data.py::test_class_names [31mERROR
test_data.py::test_column_ranges [31mERROR
test_data.py::test_kolmogorov_smirnov [31mERROR
========================================================================================================== ERRORS ==========================================================================================================
[31m[1m_____________________________________________________________________________________ ERROR at setup of test_column_presence_and_type ______________________________________________________________________________________
args = (<wandb.apis.public.Api object at 0x7f51c00c03a0>,), kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"', payload = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <wandb.apis.public.Api object at 0x7f51c00c03a0>, name = 'exercise_14/preprocessed_data.csv:latest', type = None
    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>, client = <wandb.apis.public.RetryingClient object at 0x7f51c00c0a90>, entity = 'mdunford'
project = 'exercise_14', name = 'preprocessed_data.csv:latest', attrs = None
    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           ValueError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: ValueError
[33mDuring handling of the above exception, another exception occurred:
request = <SubRequest 'data' for <Function test_column_presence_and_type>>
    @pytest.fixture(scope="session")
    def data(request):
        reference_artifact = request.config.option.reference_artifact
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
        sample_artifact = request.config.option.sample_artifact
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
>       local_path = run.use_artifact(reference_artifact).file()
[31m[1mconftest.py[39m[22m:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/sdk/wandb_run.py[39m[22m:2414: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
[31m[1m/nix/store/nv3ryyaknp2g0qyjkmg4h83flqm0c0k3-python3.8-six-1.15.0/lib/python3.8/site-packages/six.py[39m[22m:702: in reraise
    raise value.with_traceback(tb)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24: in wrapper
    return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: CommError
[31m[1m____________________________________________________________________________________________ ERROR at setup of test_class_names ____________________________________________________________________________________________
args = (<wandb.apis.public.Api object at 0x7f51c00c03a0>,), kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"', payload = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <wandb.apis.public.Api object at 0x7f51c00c03a0>, name = 'exercise_14/preprocessed_data.csv:latest', type = None
    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>, client = <wandb.apis.public.RetryingClient object at 0x7f51c00c0a90>, entity = 'mdunford'
project = 'exercise_14', name = 'preprocessed_data.csv:latest', attrs = None
    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           ValueError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: ValueError
[33mDuring handling of the above exception, another exception occurred:
request = <SubRequest 'data' for <Function test_column_presence_and_type>>
    @pytest.fixture(scope="session")
    def data(request):
        reference_artifact = request.config.option.reference_artifact
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
        sample_artifact = request.config.option.sample_artifact
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
>       local_path = run.use_artifact(reference_artifact).file()
[31m[1mconftest.py[39m[22m:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/sdk/wandb_run.py[39m[22m:2414: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
[31m[1m/nix/store/nv3ryyaknp2g0qyjkmg4h83flqm0c0k3-python3.8-six-1.15.0/lib/python3.8/site-packages/six.py[39m[22m:702: in reraise
    raise value.with_traceback(tb)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24: in wrapper
    return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: CommError
[31m[1m___________________________________________________________________________________________ ERROR at setup of test_column_ranges ___________________________________________________________________________________________
args = (<wandb.apis.public.Api object at 0x7f51c00c03a0>,), kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"', payload = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <wandb.apis.public.Api object at 0x7f51c00c03a0>, name = 'exercise_14/preprocessed_data.csv:latest', type = None
    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>, client = <wandb.apis.public.RetryingClient object at 0x7f51c00c0a90>, entity = 'mdunford'
project = 'exercise_14', name = 'preprocessed_data.csv:latest', attrs = None
    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           ValueError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: ValueError
[33mDuring handling of the above exception, another exception occurred:
request = <SubRequest 'data' for <Function test_column_presence_and_type>>
    @pytest.fixture(scope="session")
    def data(request):
        reference_artifact = request.config.option.reference_artifact
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
        sample_artifact = request.config.option.sample_artifact
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
>       local_path = run.use_artifact(reference_artifact).file()
[31m[1mconftest.py[39m[22m:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/sdk/wandb_run.py[39m[22m:2414: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
[31m[1m/nix/store/nv3ryyaknp2g0qyjkmg4h83flqm0c0k3-python3.8-six-1.15.0/lib/python3.8/site-packages/six.py[39m[22m:702: in reraise
    raise value.with_traceback(tb)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24: in wrapper
    return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: CommError
[31m[1m________________________________________________________________________________________ ERROR at setup of test_kolmogorov_smirnov _________________________________________________________________________________________
args = (<wandb.apis.public.Api object at 0x7f51c00c03a0>,), kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"', payload = 'Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <wandb.apis.public.Api object at 0x7f51c00c03a0>, name = 'exercise_14/preprocessed_data.csv:latest', type = None
    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>, client = <wandb.apis.public.RetryingClient object at 0x7f51c00c0a90>, entity = 'mdunford'
project = 'exercise_14', name = 'preprocessed_data.csv:latest', attrs = None
    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           ValueError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: ValueError
[33mDuring handling of the above exception, another exception occurred:
request = <SubRequest 'data' for <Function test_column_presence_and_type>>
    @pytest.fixture(scope="session")
    def data(request):
        reference_artifact = request.config.option.reference_artifact
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
        sample_artifact = request.config.option.sample_artifact
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
>       local_path = run.use_artifact(reference_artifact).file()
[31m[1mconftest.py[39m[22m:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/sdk/wandb_run.py[39m[22m:2414: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
[31m[1m/nix/store/nv3ryyaknp2g0qyjkmg4h83flqm0c0k3-python3.8-six-1.15.0/lib/python3.8/site-packages/six.py[39m[22m:702: in reraise
    raise value.with_traceback(tb)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/normalize.py[39m[22m:24: in wrapper
    return func(*args, **kwargs)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:761: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3443: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f51c00c05b0>
    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
[31m[1mE           wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m[1m/home/marc/.conda/envs/mlflow-acbfa4ee4aa9be3158264a605e4861dd8a78622b/lib/python3.9/site-packages/wandb/apis/public.py[39m[22m:3961: CommError
================================================================================================= short test summary info ==================================================================================================
ERROR test_data.py::test_column_presence_and_type - wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
ERROR test_data.py::test_class_names - wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
ERROR test_data.py::test_column_ranges - wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
ERROR test_data.py::test_kolmogorov_smirnov - wandb.errors.CommError: Project mdunford/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"
[31m==================================================================================================== [1m4 errors[22m in 0.97s =====================================================================================================